### PR TITLE
Added solution to HW33

### PR DESCRIPTION
**Pull request by MichaelShevchenko**

### CODE REVIEW

**Review by dariiabutorina — APPROVED** [review]
Overall, I like what I see.
Have some doubts in a few places, but that's just because you've got a different realization. 
Want to see what mentors will tell.
I don't see any strong issues, so am approving :)

UPD: I personally try to avoid the VOID type of methods, because they might not be covered as much as desired by tests. I would rather change the return type of values to the BOOLEAN or in the places we can return the specified objects it's better to do this (like in my project with adding the products etc).

**Comment by MichaelShevchenko** [comment]
наверное, ты права. я чего-то думал что приватные методы не надо тестить. но не помню источник данного тезиса)

File: src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create provided user: " + user, e);
+        }
+        addRoles(user.getRoles(), user.getId());
+        return user;
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java:62]: does user have roles_id null in create method, or they are initialized properly?

**Review by RomanDubovskyi — COMMENTED** [review]
Good job 👍
left one comment

File: src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java (hunk 1 above)
Comment (MichaelShevchenko) [reply src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java:62]: returned user have null_id due to fact that in the add method I execute insert statement with join as subquery. to make returned user have role set with proper/initialized id I can use either 
- return get(user.getId()).get() or
- user.setRoles(composeRoleSetByUserId(user.getId()));
  return user;
which one to choose?

Comment (RomanDubovskyi) [reply src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java:62]: i think second option makes more sense

Comment (MichaelShevchenko) [reply src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java:62]: chose second option. please, let me know if first option more suitable

Comment (MichaelShevchenko) [reply src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java:62]: done! I think i was uploading the changes at time you wrote your response) nice to know the prefered solution seemed more suitable for me as well

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**